### PR TITLE
Added crop scaling method

### PR DESCRIPTION
**Pull request by DarkDefender**
Added a method that scales and crop the image so that the image will
fill the whole window.

Also made viewport update respect the current scaling mode.

### CODE REVIEW

**Comment by DarkDefender** [comment]
I left out updating the bind parsing as I have simplified that in #181

File: src/viewport.h [hunk 1]
```diff
@@ -6,6 +6,21 @@
...
 
+enum scaling_mode {
+  SCALING_NONE,
+  SCALING_DOWN,
+  SCALING_FULL,
+  SCALING_CROP,
+  SCALING_MODE_COUNT
+};
+
+static const char *scaling_label[] = {
```
Comment (eXeC64) [inline src/viewport.h]: I think you want to declare this in the header, but give its value in the C file, or you get compile warnings like:

```
In file included from src/viewport.c:1:
src/viewport.h:17:20: warning: ‘scaling_label’ defined but not used [-Wunused-variable]
   17 | static const char *scaling_label[] = {
      |                    ^~~~~~~~~~~~~
```

File: src/viewport.c [hunk 2]
```diff
@@ -198,15 +198,48 @@ void imv_viewport_scale_to_window(struct imv_viewport *view, const struct imv_im
...
+  imv_viewport_center(view, image);
+  view->locked = 0;
+}
+
 void imv_viewport_set_redraw(struct imv_viewport *view)
 {
   view->redraw = 1;
 }
 
+void imv_viewport_rescale(struct imv_viewport *view, const struct imv_image *image,
```
Comment (eXeC64) [inline src/viewport.c]: This needs a declaration in the header

**Comment by DarkDefender** [comment]
I am running imv, with all my patches applied so I think this is working nicely with #186

**Comment by eXeC64** [comment]
Applied to master. Thanks!

**Pull request closed without merging**